src/dtls.c

```diff
@@ -285,10 +285,13 @@ static int ParseClientHello(const byte* input, word32 helloSz, WolfSSL_CH* ch)
     if (idx > helloSz - OPAQUE8_LEN)
         return BUFFER_ERROR;
     idx += ReadVector8(input + idx, &ch->compression);
-    if (idx > helloSz - OPAQUE16_LEN)
-        return BUFFER_ERROR;
-    idx += ReadVector16(input + idx, &ch->extension);
-    if (idx > helloSz)
+    if (idx < helloSz - OPAQUE16_LEN) {
```

Comment:
It looks like ParseClientHello is also used with DTLS 1.3. Are extensions optional with the newer versions of DTLS? Briefly digging through rfc9147 this was not clear to me. Maybe we are doing additional extension sanity checks specific to DTLS 1.3 at a later time after this parsing section?

Reply:
Extensions are mandatory in 1.3 for version negotiation but if we don't receive any extensions then this connection simply won't negotiate 1.3. This is checked in `TlsCheckSupportedVersion`.